src/text/parsers/decimal.rs

```diff
@@ -107,11 +112,11 @@ fn decimal_from_text_components(
     let mut exponent = i64::from_str(&sanitized).expect("parsing exponent as i64 failed");
     // Reduce the exponent by the number of digits that follow the decimal point
     exponent -= digits_after_dot.chars().filter(|c| c.is_digit(10)).count() as i64;
-    Decimal::new(coefficient, exponent)
+    Ok(("", Decimal::new(coefficient, exponent)))
```

Comment:
🗺️ Now that we're returning `IonParseResult` everywhere, we can't just return `Decimal`. We have to use `nom`'s success data type: `Ok(...)` containing a tuple of the remaining input (a `&str`) and the value we parsed (a `Decimal`).